infra/ci_fuzz.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module used by CI tools in order to interact with fuzzers.
+This module helps CI tools do the following
+  1. build fuzzers
+  2. run fuzzers
+Eventually it will be used to help CI tools determine which fuzzers to run.
+"""
+
+import argparse
+import os
+import tempfile
+
+import build_specified_commit
+
+
+def main():
+  """Connects Fuzzers with CI tools."""
+  parser = argparse.ArgumentParser(
+      description='Help CI tools manage specific fuzzers')
+
+  subparsers = parser.add_subparsers(dest='command')
+  build_fuzzer_parser = subparsers.add_parser(
+      'build_fuzzers', help='Build fuzzers')
+  build_fuzzer_parser.add_argument('project_name')
+  build_fuzzer_parser.add_argument('repo_name')
+  build_fuzzer_parser.add_argument('commit_sha')
+
+  run_fuzzer_parser = subparsers.add_parser(
+      'run_fuzzers', help='Run a specific projects fuzzers')
+  run_fuzzer_parser.add_argument('project_name')
+  args = parser.parse_args()
+
+  if args.command == 'build_fuzzers':
+    return build_fuzzers(args)
+  elif args.command == 'run_fuzzer':
+    return run_fuzzers(args)
+  print('Invalid argument option, use  build_fuzzers or run_fuzzer')
+  return 1
+
+
+def build_fuzzers(args):
+  """Builds all of the fuzzers for a specific OSS-Fuzz project."""
+  # Change to oss-fuzz main directory so helper.py runs correctly
+  if os.getcwd() != os.path.dirname(
+      os.path.dirname(os.path.realpath(__file__))):
+    os.chdir(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+  with tempfile.TemporaryDirectory() as tmp_dir:
+    print("Detecting repo with name: " + args.repo_name)
+    return build_specified_commit.build_fuzzer_from_commit(
```

Comment:
It's pretty ugly that almost every function returns 1 on failure and 0 on success just so that the return value can bubble all the way back to the process exit. I think we will need to change this in another CL.

Reply:
Noted.